django/contrib/gis/templates/gis/admin/openlayers.js

```diff
@@ -12,98 +12,98 @@ OpenLayers.Projection.addTransform("EPSG:4326", "EPSG:3857", OpenLayers.Layer.Sp
 {{ module }}.is_point = {{ is_point|yesno:"true,false" }};
 {% endblock %}
 {{ module }}.get_ewkt = function(feat){
-  return 'SRID={{ srid|unlocalize }};' + {{ module }}.wkt_f.write(feat);
+    return 'SRID={{ srid|unlocalize }};' + {{ module }}.wkt_f.write(feat);
 };
 {{ module }}.read_wkt = function(wkt){
-  // OpenLayers cannot handle EWKT -- we make sure to strip it out.
-  // EWKT is only exposed to OL if there's a validation error in the admin.
-  var match = {{ module }}.re.exec(wkt);
-  if (match){wkt = match[1];}
-  return {{ module }}.wkt_f.read(wkt);
+    // OpenLayers cannot handle EWKT -- we make sure to strip it out.
+    // EWKT is only exposed to OL if there's a validation error in the admin.
+    var match = {{ module }}.re.exec(wkt);
+    if (match){wkt = match[1];}
+    return {{ module }}.wkt_f.read(wkt);
 };
 {{ module }}.write_wkt = function(feat){
-  if ({{ module }}.is_collection){ {{ module }}.num_geom = feat.geometry.components.length;}
-  else { {{ module }}.num_geom = 1;}
-  document.getElementById('{{ id }}').value = {{ module }}.get_ewkt(feat);
+    if ({{ module }}.is_collection){ {{ module }}.num_geom = feat.geometry.components.length;}
+    else { {{ module }}.num_geom = 1;}
+    document.getElementById('{{ id }}').value = {{ module }}.get_ewkt(feat);
 };
 {{ module }}.add_wkt = function(event){
-  // This function will sync the contents of the `vector` layer with the
-  // WKT in the text field.
-  if ({{ module }}.is_collection){
-    var feat = new OpenLayers.Feature.Vector(new OpenLayers.Geometry.{{ geom_type }}());
-    for (var i = 0; i < {{ module }}.layers.vector.features.length; i++){
-      feat.geometry.addComponents([{{ module }}.layers.vector.features[i].geometry]);
-    }
-    {{ module }}.write_wkt(feat);
-  } else {
-    // Make sure to remove any previously added features.
-    if ({{ module }}.layers.vector.features.length > 1){
-      old_feats = [{{ module }}.layers.vector.features[0]];
-      {{ module }}.layers.vector.removeFeatures(old_feats);
-      {{ module }}.layers.vector.destroyFeatures(old_feats);
+    // This function will sync the contents of the `vector` layer with the
+    // WKT in the text field.
+    if ({{ module }}.is_collection){
```

Comment:
I just pushed changes to replace JSHint with ESLint.  Running "npm test" will print out a number of warnings but the return code is 0 because no errors are returned.  I downgraded a number of default ESLint [rules][] from errors to warnings and made a number of changes to the JS files to ensure there are no errors returned.

If this change might be too drastic to make without tests, we could hold off or downgrade more errors to warnings and uncommit some of the code changes I made.

[rules]: https://github.com/django/django/pull/4577/files#diff-1dc6ee56b778cd91e0327b52aaeaa1b9R2